test/static.test.js

```diff
@@ -306,6 +306,44 @@ t.test('register /static/', t => {
   })
 })
 
+t.test('payload.filename is set', t => {
+  t.plan(2)
+
+  const pluginOptions = {
+    root: path.join(__dirname, '/static'),
+    prefix: '/static/'
+  }
+  const fastify = Fastify()
+  let gotFilename
+  fastify.register(fastifyStatic, pluginOptions)
+  fastify.addHook('onSend', function (req, reply, payload, next) {
+    gotFilename = payload.filename
```

Comment:
You can also add a typecheck here, something like `t.is(typeof payload.filename, 'string')`.

Reply:
I've instead added a check for `typeof gotFilename` inside the `simple.concat` callback, added a 404 test.  This way it verifies payload.filename is string on success, undefined on 404.